sale_allotment/models/stock_move.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2015 Odoo.com.
+#    Copyright (C) 2015 Openies.com.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def _picking_assign(self, cr, uid, move_ids, procurement_group,
+                        location_from, location_to, context=None):
+        partner_dict = {}
+        vals = super(StockMove, self)._picking_assign(
+            cr, uid, move_ids, procurement_group, location_from, location_to,
+            context=context)
+        move_rec = self.browse(cr, uid, move_ids, context=context)
+        for move in move_rec:
+            if not partner_dict.get(move.partner_id.id, False):
+                partner_dict.update({move.partner_id.id: [move.id]})
+            else:
+                partner_dict.get(move.partner_id.id).append(move.id)
+        pick_obj = self.pool.get('stock.picking')
+        for key in partner_dict.keys():
+            move = move_rec[0]
+            if partner_dict.keys().index(key) == 0:
```

Comment:
What is this line supposed to do? It looks like you rely on the order of the keys of the dictionary which is unsafe when using regular dictionaries. You need collections.OrderedDict for that.